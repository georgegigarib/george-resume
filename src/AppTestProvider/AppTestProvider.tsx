import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
