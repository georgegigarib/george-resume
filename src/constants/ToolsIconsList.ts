import JsIcon from '@/assets/icons/tools/Javascript.svg'

export type ToolInfo = {
  name: string
  icon: IconName
  color: string
  bgColor: string
  darkBgColor?: string
  borderColor: string
  darkBorderColor?: string
  darkColor?: string
  svg?: string
  custom?: boolean
}

export type ToolIconProps = {
  tool: IconName
  size?: '5x' | '4x' | '3x' | '2x' | 'xl' | 'lg' | 'md' | 'sm' | 'xs'
}

export enum IconName {
  React = 'react',
  ReactNative = 'reactnative',
  TailwindCSS = 'tailwindcss',
  JavaScript = 'js',
  TypeScript = 'ts-circle',
  VueJs = 'vuejs',
  NextJS = 'nextjs',
  NodeJS = 'nodejs',
  HTML = 'html',
  CSS = 'css',
  Express = 'express',
  Spotify = 'spotify',
  Git = 'git',
  Redis = 'redis',
  Bun = 'bun',
  Docker = 'docker',
  Pinia = 'pinia',
  Sass = 'sass',
  Bootstrap = 'bootstrap',
  Angular = 'angular',
  C = 'c',
  CSharp = 'csharp',
  Django = 'django',
  FramerMotion = 'framer-motion',
  Go = 'go',
  Graphql = 'graphql',
  Javalanguage = 'java',
  Laravel = 'laravel',
  Mui = 'mui',
  Jest = 'jest',
  Vitest = 'vitest',
  Cypress = 'cypress',
  MySQL = 'mysql',
  MongoDb = 'mongodb',
  Php = 'php',
  Postgresql = 'postgresql',
  Python = 'python',
  Redux = 'redux',
  Zustand = 'zustand',
  Svelte = 'svelte',
}

export const toolsIconsList: ToolInfo[] = [
  {
    name: 'React',
    icon: IconName.React,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'React',
    icon: IconName.ReactNative,
    color: '#30CDF9FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#61DAFB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Tailwind CSS',
    icon: IconName.TailwindCSS,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#38BDF8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'JavaScript',
    icon: IconName.JavaScript,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
    custom: true,
    svg: JsIcon,
  },
  {
    name: 'Jest',
    icon: IconName.Jest,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MySQL',
    icon: IconName.MySQL,
    color: '#09A1E2FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#38BDF8',
  },
  {
    name: 'Vitest',
    icon: IconName.Vitest,
    color: '#F7DF1E',
    bgColor: '#FFFFFFFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#F7DF1E',
  },
  {
    name: 'Cypress',
    icon: IconName.Cypress,
    color: '#000000FF',
    bgColor: '#B4B4B4FF',
    darkBgColor: '#BABABAFF',
    borderColor: '#C7B008FF',
  },
  {
    name: 'TypeScript',
    icon: IconName.TypeScript,
    color: '#3178C6',
    bgColor: '#FAFAFAFF',
    borderColor: '#3178C6',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Next.js',
    icon: IconName.NextJS,
    color: '#0000000',
    bgColor: '#FAFAFAFF',
    borderColor: '#555555FF',
    darkBgColor: '#3E3E3EFF',
    darkColor: '#FFFFFF',
    darkBorderColor: '#BFBFBFFF',
  },
  {
    name: 'Node.js',
    icon: IconName.NodeJS,
    color: '#539E43',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#539E43',
  },
  {
    name: 'HTML',
    icon: IconName.HTML,
    color: '#E44D26',
    bgColor: '#FAFAFAFF',
    borderColor: '#E44D26',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'CSS',
    icon: IconName.CSS,
    color: '#264DE4',
    bgColor: '#FAFAFAFF',
    borderColor: '#264DE4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Express',
    icon: IconName.Express,
    color: '#333333',
    bgColor: '#FAFAFAFF',
    borderColor: '#CCCCCC',
  },
  {
    name: 'Spotify',
    icon: IconName.Spotify,
    color: '#1DB954',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#1DB954',
  },
  { name: 'Git', icon: IconName.Git, color: '#F05032', bgColor: '#F5F5F5', borderColor: '#F05032' },
  {
    name: 'Redis',
    icon: IconName.Redis,
    color: '#D82C20',
    bgColor: '#FAFAFAFF',
    borderColor: '#D82C20',
  },
  {
    name: 'Bun',
    icon: IconName.Bun,
    darkColor: '#ffffff',
    color: '#333333',
    bgColor: '#FAFAFAFF',
    darkBorderColor: '#bababa',
    borderColor: '#686767FF',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Docker',
    icon: IconName.Docker,
    color: '#2496ED',
    bgColor: '#FAFAFAFF',
    borderColor: '#2496ED',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Pinia',
    icon: IconName.Pinia,
    color: '#F0C723FF',
    darkBgColor: '#2C2C2C',
    bgColor: '#FAFAFAFF',
    borderColor: '#FFD939',
  },
  {
    name: 'Sass',
    icon: IconName.Sass,
    color: '#CC6699',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#CC6699',
  },
  {
    name: 'Bootstrap',
    icon: IconName.Bootstrap,
    color: '#A877F3FF',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#7952B3',
  },
  {
    name: 'Angular',
    icon: IconName.Angular,
    color: '#DD0031',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#DD0031',
  },
  {
    name: 'C',
    icon: IconName.C,
    color: '#00599C',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#00599C',
  },
  {
    name: 'C#',
    icon: IconName.CSharp,
    color: '#239120',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#239120',
  },
  {
    name: 'Django',
    icon: IconName.Django,
    color: '#092E20',
    bgColor: '#FAFAFAFF',
    borderColor: '#092E20',
  },
  {
    name: 'Framer Motion',
    icon: IconName.FramerMotion,
    color: '#EC5990',
    bgColor: '#FAFAFAFF',
    darkBgColor: '#2C2C2C',
    borderColor: '#EC5990',
  },
  {
    name: 'Go',
    icon: IconName.Go,
    color: '#00ADD8',
    bgColor: '#FAFAFAFF',
    borderColor: '#00ADD8',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'GraphQL',
    icon: IconName.Graphql,
    color: '#E10098',
    bgColor: '#FAFAFAFF',
    borderColor: '#E10098',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'java',
    icon: IconName.Javalanguage,
    color: '#007396',
    bgColor: '#FAFAFAFF',
    borderColor: '#007396',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Laravel',
    icon: IconName.Laravel,
    color: '#FF2D20',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF2D20',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Material UI',
    icon: IconName.Mui,
    color: '#0081CB',
    bgColor: '#FAFAFAFF',
    borderColor: '#0081CB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'MongoDB',
    icon: IconName.MongoDb,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Vue Js',
    icon: IconName.VueJs,
    color: '#47A248',
    bgColor: '#FAFAFAFF',
    borderColor: '#47A248',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PHP',
    icon: IconName.Php,
    color: '#959AE1FF',
    bgColor: '#FAFAFAFF',
    borderColor: '#777BB4',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'PostgreSQL',
    icon: IconName.Postgresql,
    color: '#336791',
    bgColor: '#FAFAFAFF',
    borderColor: '#336791',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Python',
    icon: IconName.Python,
    color: '#3776AB',
    bgColor: '#FAFAFAFF',
    borderColor: '#3776AB',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Redux',
    icon: IconName.Redux,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Zustand',
    icon: IconName.Zustand,
    color: '#764ABC',
    bgColor: '#FAFAFAFF',
    borderColor: '#764ABC',
    darkBgColor: '#2C2C2C',
  },
  {
    name: 'Svelte',
    icon: IconName.Svelte,
    color: '#FF3E00',
    bgColor: '#FAFAFAFF',
    borderColor: '#FF3E00',
    darkBgColor: '#2C2C2C',
  },
]
