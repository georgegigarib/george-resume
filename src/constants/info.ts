export const WORK_EMAIL = 'georgegil.work@gmail.com'
export const SOCIAL_LINKS = {
  github: 'https://github.com/georgegigarib',
  linkedin: 'https://www.linkedin.com/in/georgegg/',
  instagram: 'URL_ADDRESS.com/GeorgeGil',
}
