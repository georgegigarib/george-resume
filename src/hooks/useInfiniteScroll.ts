import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
