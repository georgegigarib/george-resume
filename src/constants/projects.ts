import image1 from '@/assets/images/projects/movie-app.gif'
import image2 from '@/assets/images/projects/mobile-2.jpeg'
import image3 from '@/assets/images/projects/mobile-1.jpg'
import image4 from '@/assets/images/projects/desktop-app-1.png'
import image5 from '@/assets/images/projects/desktop-app-2.png'
import image6 from '@/assets/images/projects/desktop-app-3.png'
import { IconName } from './ToolsIconsList'

export type ProjectInfoType = {
  src: string
  title: string
  description: string
  url: string
  tools: IconName[]
}

export const mobileProjects: ProjectInfoType[] = [
  {
    src: image1,
    title: 'Movie App',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto y debo alargarlo mas y mas y mas',
    url: 'https://movie-app.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.JavaScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS, IconName.JavaScript]
  },
  {
    src: image3,
    title: 'Proyecto Mobile 2',
    description:
      ' Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript, IconName.JavaScript]
  },
  {
    src: image1,
    title: 'Proyecto Mobile 2',
    description:
      'Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto, Aplicación para explorar películas y series. aqui estoy dejando una descripcion un poco mas lagra para pobar que pasa sies muy largo el texto',
    url: 'https://project2.example.com',
    tools: [IconName.React, IconName.TypeScript]
  },
  {
    src: image2,
    title: 'Proyecto Mobile 1',
    description: 'Proyecto con animaciones personalizadas.',
    url: 'https://project1.example.com',
    tools: [IconName.HTML, IconName.CSS]
  }
]

export const desktopProjects: ProjectInfoType[] = [
  {
    src: image6,
    title: 'Proyecto Desktop 1',
    description: 'Dashboard avanzado.',
    url: 'https://desktop1.example.com',
    tools: [
      IconName.React,
      IconName.TailwindCSS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description:
      'Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada. Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TypeScript,
      IconName.JavaScript,
      IconName.CSS,
      IconName.NextJS,
      IconName.NodeJS,
      IconName.TailwindCSS,
      IconName.Redis,
      IconName.Git
    ]
  },
  {
    src: image4,
    title: 'Proyecto Desktop 3',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image5,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.NextJS, IconName.NodeJS, IconName.TailwindCSS, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  },
  {
    src: image6,
    title: 'Proyecto Desktop 2',
    description: 'Aplicación de escritorio personalizada.',
    url: 'https://desktop2.example.com',
    tools: [IconName.React, IconName.TailwindCSS, IconName.TypeScript, IconName.Redis, IconName.Git]
  }
]
