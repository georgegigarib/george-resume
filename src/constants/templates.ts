import { IconName } from './ToolsIconsList'

export interface Template {
  tools: IconName[]
  url: string
  hint: string
  bgColor: string
  description: string
}

export const devTemplates: Template[] = [
  {
    tools: [IconName.TypeScript, IconName.React, IconName.TailwindCSS, IconName.Bun, IconName.CSS],
    url: 'https://github.com/example/fast-render-template',
    hint: 'fast rendering',
    bgColor: '#919469FF',
    description: 'A lightweight template designed for fast rendering with modern tools.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.NextJS,
      IconName.TailwindCSS,
      IconName.Docker,
      IconName.React
    ],
    url: 'https://github.com/example/next-ssr-template',
    hint: 'server rendering',
    bgColor: '#B99E4CFF',
    description: 'A server-side rendering starter template using Next.js and Docker.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.VueJs,
      IconName.TailwindCSS,
      IconName.Pinia,
      IconName.Docker
    ],
    url: 'https://github.com/example/vue-pinia-template',
    hint: 'vue simplicity',
    bgColor: '#6E8C9AFF',
    description: 'A clean Vue.js starter with Pinia and TailwindCSS for simplicity and scalability.'
  },
  {
    tools: [IconName.JavaScript, IconName.Angular, IconName.Sass, IconName.Docker, IconName.NodeJS],
    url: 'https://github.com/example/angular-enterprise-template',
    hint: 'enterprise apps',
    bgColor: '#A26B5AFF',
    description: 'A robust template for building enterprise-grade Angular applications.'
  },
  {
    tools: [IconName.TypeScript, IconName.React, IconName.Mui, IconName.Graphql, IconName.Django],
    url: 'https://github.com/example/react-graphql-template',
    hint: 'full stack',
    bgColor: '#877F95FF',
    description: 'Full-stack development made easy with React, MUI, and GraphQL backend.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.Svelte,
      IconName.TailwindCSS,
      IconName.Python,
      IconName.Postgresql
    ],
    url: 'https://github.com/example/svelte-minimal-template',
    hint: 'minimalist setup',
    bgColor: '#687969FF',
    description:
      'A minimalist yet powerful template using Svelte and Python with a PostgreSQL database.'
  },
  {
    tools: [IconName.JavaScript, IconName.React, IconName.Redux, IconName.Bootstrap, IconName.Php],
    url: 'https://github.com/example/ecommerce-template',
    hint: 'ecommerce base',
    bgColor: '#908C63FF',
    description: 'A ready-to-use base for building eCommerce solutions with React and PHP.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.React,
      IconName.Graphql,
      IconName.MongoDb,
      IconName.Docker
    ],
    url: 'https://github.com/example/data-intensive-template',
    hint: 'data intensive',
    bgColor: '#96659FFF',
    description: 'A powerful template for data-intensive applications using MongoDB and GraphQL.'
  },
  {
    tools: [
      IconName.TypeScript,
      IconName.React,
      IconName.TailwindCSS,
      IconName.FramerMotion,
      IconName.NodeJS
    ],
    url: 'https://github.com/example/firebase-react-template',
    hint: 'real-time apps',
    bgColor: '#3E848DFF',
    description: 'A template for real-time apps with Firebase backend and React frontend.'
  }
]
