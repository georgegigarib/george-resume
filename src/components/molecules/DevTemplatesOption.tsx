import React, { useState } from 'react'
import clsx from 'clsx'
import TechnologyBadge from '@/components/atoms/TechnologyBadge'
import Tooltip from '@mui/material/Tooltip'
import type { Template } from '@/constants/templates'
import { useIsMobile } from '@/hooks/useIsMobile'
import { useTranslation } from 'react-i18next'
import LinkIcon from '@mui/icons-material/Link'
import ExpandMore from '@mui/icons-material/ExpandMore'
import ExpandLess from '@mui/icons-material/ExpandLess'
import { motion } from 'framer-motion'

interface DevTemplatesOptionProps {
  option: Template
}

const DevTemplatesOption: React.FC<DevTemplatesOptionProps> = ({ option }) => {
  const [isExpanded, setIsExpanded] = useState(false)
  const { t } = useTranslation()
  const isMobile = useIsMobile()

  const MAX_VISIBLE_BADGES = isMobile ? 1 : 3
  const remainingTools = option.tools.length - MAX_VISIBLE_BADGES

  const handleContainerClick = () => {
    if (!isExpanded && remainingTools > 0) {
      setIsExpanded(true)
    }
  }

  const handleExpand = () => {
    setIsExpanded(true)
  }

  const handleCollapse = () => {
    setIsExpanded(false)
  }

  return (
    <div
      className={clsx('flex w-full rounded-lg border border-gray-300 flex-nowrap mt-4 relative')}
      data-testid="template-option"
    >
      <div className="overflow-hidden max-w-[86%]">
        <motion.div
          initial={{ height: 'auto' }}
          animate={{ height: isExpanded ? 'auto' : '50px' }}
          transition={{ duration: 0.3 }}
          className="whitespace-nowrap shadow-inner flex mt-1 ml-2 flex-col"
          onClick={handleContainerClick}
        >
          <div className="flex flex-wrap">
            {option.tools.map((tool, index) => (
              <div className="my-1 mb-2" data-testid="tec-badge" key={index}>
                <TechnologyBadge tool={tool} size={isMobile ? 'sm' : 'lg'} />
              </div>
            ))}
          </div>

          <div className="overflow-auto text-wrap my-2 max-h-20 description-scrollbar">{t(option.description)}</div>
        </motion.div>

        <motion.div
          className={clsx('absolute top-3 right-2 cursor-pointer')}
          whileHover={{ scale: 1.05 }}
          onClick={() => window.open(option.url, '_blank')}
        >
          <div className="hover:bg-gray-700 rounded-full p-1  flex items-center justify-center">
            <LinkIcon fontSize="medium" />
          </div>
        </motion.div>
      </div>

      {remainingTools > 0 && !isExpanded && (
        <motion.div
          className="absolute -bottom-3 right-3 cursor-pointer bg-app-grey rounded-full"
          whileHover={{ scale: 1.05 }}
          onClick={handleExpand}
        >
          <Tooltip title={t('dev.templates.modal.showAllTools')} arrow>
            <span className="ml-auto cursor-pointer hover:opacity-70 flex items-center">
              <div className="flex items-center justify-center rounded-full text-sm ">
                <span className="text-white rounded-full flex flex-nowrap">
                  <ExpandMore fontSize="medium" />
                </span>
              </div>
            </span>
          </Tooltip>
        </motion.div>
      )}

      {isExpanded && (
        <>
          <motion.div
            className="absolute -bottom-3 right-3 cursor-pointer bg-blue-950 rounded-full"
            whileHover={{ scale: 1.05 }}
            onClick={handleCollapse}
          >
            <Tooltip title="See less" arrow>
              <div className="flex items-center justify-center rounded-full text-white  hover:bg-gray-700 -bottom-4">
                <ExpandLess fontSize="medium" />
              </div>
            </Tooltip>
          </motion.div>
        </>
      )}
    </div>
  )
}

export default DevTemplatesOption
