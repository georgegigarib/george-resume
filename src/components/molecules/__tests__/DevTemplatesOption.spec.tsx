import { render } from '@testing-library/react'
import { describe, expect, it, vi, beforeEach, Mock } from 'vitest'
import DevTemplatesOption from '@/components/molecules/DevTemplatesOption'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'
import { Template } from '@/constants/templates'
import { useIsMobile } from '@/hooks/useIsMobile'
import { IconName } from '@/constants/ToolsIconsList'

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}))

const mockTemplate: Template = {
  tools: [IconName.React, IconName.TypeScript, IconName.TailwindCSS],
  url: 'https://example.com',
  hint: 'example hint',
  bgColor: '#FF4500',
  description: 'Example description',
}

describe('DevTemplatesOption component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  const renderComponent = (isMobile: boolean, option: Template) => {
    ;(useIsMobile as Mock).mockReturnValue(isMobile)
    return render(
      <AppTestProvider>
        <DevTemplatesOption option={option} />
      </AppTestProvider>
    )
  }

  it('matches the snapshot for desktop view', () => {
    const { asFragment } = renderComponent(false, mockTemplate)

    expect(asFragment()).toMatchSnapshot()
  })

  it('matches the snapshot for mobile view', () => {
    const { asFragment } = renderComponent(true, mockTemplate)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the correct number of technology badges for desktop view', () => {
    const { getAllByTestId } = renderComponent(false, mockTemplate)
    expect(getAllByTestId('tec-badge').length).toBe(3)
  })

  it('renders the correct number of technology badges for mobile view', () => {
    const { getAllByTestId } = renderComponent(true, mockTemplate)
    expect(getAllByTestId('tec-badge').length).toBe(3)
  })
})
