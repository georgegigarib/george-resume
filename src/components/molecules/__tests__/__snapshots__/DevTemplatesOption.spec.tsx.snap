// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DevTemplatesOption component > matches the snapshot for desktop view 1`] = `
<DocumentFragment>
  <div
    class="flex w-full rounded-lg border border-gray-300 flex-nowrap mt-4 relative"
    data-testid="template-option"
  >
    <div
      class="overflow-hidden max-w-[86%]"
    >
      <div
        class="whitespace-nowrap shadow-inner flex mt-1 ml-2 flex-col"
        style="height: auto;"
      >
        <div
          class="flex flex-wrap"
        >
          <div
            class="my-1 mb-2"
            data-testid="tec-badge"
          >
            <div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
              >
                <i
                  class="ci ci-react ci-lg"
                />
                <span
                  class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                >
                  React
                </span>
              </div>
            </div>
          </div>
          <div
            class="my-1 mb-2"
            data-testid="tec-badge"
          >
            <div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
              >
                <i
                  class="ci ci-ts-circle ci-lg"
                />
                <span
                  class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                >
                  TypeScript
                </span>
              </div>
            </div>
          </div>
          <div
            class="my-1 mb-2"
            data-testid="tec-badge"
          >
            <div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1w1q3jt-MuiChip-root"
              >
                <i
                  class="ci ci-tailwind ci-lg"
                />
                <span
                  class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                >
                  Tailwind CSS
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="overflow-auto text-wrap my-2 max-h-20 description-scrollbar"
        />
      </div>
      <div
        class="absolute top-3 right-2 cursor-pointer"
      >
        <div
          class="hover:bg-gray-700 rounded-full p-1  flex items-center justify-center"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
            data-testid="LinkIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DevTemplatesOption component > matches the snapshot for mobile view 1`] = `
<DocumentFragment>
  <div
    class="flex w-full rounded-lg border border-gray-300 flex-nowrap mt-4 relative"
    data-testid="template-option"
  >
    <div
      class="overflow-hidden max-w-[86%]"
    >
      <div
        class="whitespace-nowrap shadow-inner flex mt-1 ml-2 flex-col"
        style="height: auto;"
      >
        <div
          class="flex flex-wrap"
        >
          <div
            class="my-1 mb-2"
            data-testid="tec-badge"
          >
            <div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-7lm3m0-MuiChip-root"
              >
                <i
                  class="ci ci-react ci-sm"
                />
                <span
                  class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                >
                  React
                </span>
              </div>
            </div>
          </div>
          <div
            class="my-1 mb-2"
            data-testid="tec-badge"
          >
            <div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1izr7bw-MuiChip-root"
              >
                <i
                  class="ci ci-ts-circle ci-sm"
                />
                <span
                  class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                >
                  TypeScript
                </span>
              </div>
            </div>
          </div>
          <div
            class="my-1 mb-2"
            data-testid="tec-badge"
          >
            <div>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1w1q3jt-MuiChip-root"
              >
                <i
                  class="ci ci-tailwind ci-sm"
                />
                <span
                  class="MuiChip-label MuiChip-labelMedium css-3uu4gz-MuiChip-label"
                >
                  Tailwind CSS
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="overflow-auto text-wrap my-2 max-h-20 description-scrollbar"
        />
      </div>
      <div
        class="absolute top-3 right-2 cursor-pointer"
      >
        <div
          class="hover:bg-gray-700 rounded-full p-1  flex items-center justify-center"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
            data-testid="LinkIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1M8 13h8v-2H8zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="absolute -bottom-3 right-3 cursor-pointer bg-app-grey rounded-full"
    >
      <span
        aria-label="Show all tools"
        class="ml-auto cursor-pointer hover:opacity-70 flex items-center"
        data-mui-internal-clone-element="true"
      >
        <div
          class="flex items-center justify-center rounded-full text-sm "
        >
          <span
            class="text-white rounded-full flex flex-nowrap"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </div>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
